src/builtins/exit.c

```diff
@@ -1,14 +1,51 @@
 #include <minishell.h>
 
-int	ft_exit(char **args, t_minishell *shell)
+static void exit_err_msg(char *arg)
 {
-	int	status;
+	ft_putstr_fd(SHELDON, STDERR_FILENO);
+	ft_putstr_fd(": exit: ", STDERR_FILENO);
+	ft_putstr_fd(arg, STDERR_FILENO);
+	ft_putendl_fd(": numeric argument required", STDERR_FILENO);
+}
+
+static int numeric(char *arg)
+{
+	size_t i;
+
+	i = 0;
+	while (arg[i] && ft_iswhitespace(arg[i]))
+		i++;
+	while (arg[i] && arg[i] == '0')
+		i++;
+	if (arg[i] == '-')
+		i++;
+	while (arg[i] && arg[i] != '#' && arg[i] != ';')
+	{
+		if (!ft_isdigit(arg[i]))
+			return (0);
+		i++;
+	}
+	return (1);
+}
 
-	status = EXIT_SUCCESS;
-	(void)shell;
+int	ft_exit(char **args, t_minishell *shell)
+{
 	if (args[1] != NULL)
 	{
-		status = atoi(args[1]) % 256;
+		if (!numeric(args[1]))
+		{
+			shell->status = E_UNKNOWN;
+			exit_err_msg(args[1]);
+		}
+		else
+		{
+			shell->status = (t_status) ft_atoi(args[1]) % 256;
+			if (0 > shell->status)
```

Comment:
Als je hier toch de check doet, kun je hem ook gelijk assignen

Reply:
Ik weet niet zeker wat er moet gebeuren maar miss is het makkelijker om de status eerst in een tmp op te slaan, en wellicht ft_atoi te herschrijven naar iest dat wel goede error codes kan genereren. wellicht is strtol wel een goede optie?